prow/slack/reporter/reporter.go

```diff
@@ -46,7 +47,7 @@ func channel(cfg *config.SlackReporter, pj *v1.ProwJob) string {
 }
 
 func (sr *slackReporter) Report(pj *v1.ProwJob) ([]*v1.ProwJob, error) {
-	config := sr.config()
+	config := sr.config(pj.Spec.Refs)
```

Comment:
We need a nilcheck here, if refs is nil and there is no global default, config, will be nil

Reply:
I am now returning an `error` to make it clear when a slack reporter config cannot be found.
```golang
config, err := sr.config(pj.Spec.Refs)
...
```